modules/wolfvision/eye14.rb

```diff
@@ -0,0 +1,164 @@
+# encoding: ASCII-8BIT
+# frozen_string_literal: true
+
+module Wolfvision; end
+
+# Documentation: https://www.wolfvision.com/wolf/protocol_command_wolfvision/protocol/commands_eye-14.pdf
+
+class Wolfvision::Eye14
+    include ::Orchestrator::Constants
+    include ::Orchestrator::Transcoder
+
+    # Discovery Information
+    tcp_port 50915 # Need to go through an RS232 gatway
+    descriptive_name 'WolfVision EYE-14'
+    generic_name :Camera
+
+    # Communication settings
+    tokenize indicator: /\x00|\x01|/, callback: :check_length
+    delay between_sends: 150
+
+    def on_load
+        self[:zoom_max] = 3923
+        self[:iris_max] = 4094
+        self[:zoom_min] = self[:iris_min] = 0
+        on_update
+    end
+
+    def on_update
+    end
+
+    def on_unload
+    end
+
+    def connected
+        schedule.every('60s') do
+            logger.debug "-- Polling Sony Camera"
+            power? do
+                if self[:power] == On
+                    zoom?
+                    iris?
+                    autofocus?
+                end
+            end
+        end
+    end
+
+    def disconnected
+        # Disconnected will be called before connect if initial connect fails
+        schedule.clear
+    end
+
+    def power(state)
+        target = is_affirmative?(state)
+        self[:power_target] = target
+
+        # Execute command
+        logger.debug { "Target = #{target} and self[:power] = #{self[:power]}" }
+        if target == On && self[:power] != On
+            send_cmd("\x30\x01\x01", name: :power_cmd)
+        elsif target == Off && self[:power] != Off
+            send_cmd("\x30\x01\x00", name: :power_cmd)
+        end
+    end
+
+    # uses only optical zoom
+    def zoom(position)
+        val = in_range(position, self[:zoom_max], self[:zoom_min])
+        self[:zoom_target] = val
+        val = format("%04X", val)
+        logger.debug { "position in decimal is #{position} and hex is #{val}" }
+        send_cmd("\x20\x02#{hex_to_byte(val)}", name: :zoom_cmd)
+    end
+
+    def zoom?
+        send_inq("\x20\x00", priority: 0, name: :zoom_inq)
+    end
+
+    # set autofocus to on
+    def autofocus
+        send_cmd("\x31\x01\x01", name: :autofocus_cmd)
+    end
+
+    def autofocus?
+        send_inq("\x31\x00", priority: 0, name: :autofocus_inq)
+    end
+
+    def iris(position)
+        val = in_range(position, self[:iris_max], self[:iris_min])
+        self[:iris_target] = val
+        val = format("%04X", val)
+        logger.debug { "position in decimal is #{position} and hex is #{val}" }
+        send_cmd("\x22\x02#{hex_to_byte(val)}", name: :iris_cmd)
+    end
```

Comment:
I think instead of using binary strings when building commands, you should just use ASCII
There is a helper that'll do the binary conversion for you: https://developer.acaprojects.com/#/driver-development/device-drivers?id=transmitting-a-request

```ruby
    def iris(position)
        val = in_range(position, self[:iris_max], self[:iris_min])
        self[:iris_target] = val
        logger.debug { "position in decimal is #{val}" }
        send_cmd("2202#{val.to_s(16).rjust(2, '0')}", name: :iris_cmd)
    end

    # ...

    def send_cmd(cmd, **options)
        req = "01#{cmd}"
        logger.debug { "tell -- 0x#{req} -- #{options[:name]}" }
        options[:hex_string] = true
        send(req, options)
    end
```

Means we don't need to call `byte_to_hex` so often and the strings are slightly more readable, as the don't require the \x anymore